src/OpenToolkit.Mathematics/Geometry/Box2.cs

```diff
@@ -19,140 +19,264 @@ namespace OpenToolkit.Mathematics
     [StructLayout(LayoutKind.Sequential)]
     public struct Box2 : IEquatable<Box2>
     {
+        private Vector2 _min;
         /// <summary>
-        /// The left boundary of the structure.
+        /// The minimum boundary of the structure.
         /// </summary>
-        public float Left;
+        public Vector2 Min
+        {
+            get => _min;
+            set
+            {
+                if (value.X > _max.X)
+                {
+                    _min.X = _max.X;
+                    _max.X = value.X;
+                }
+                else
+                {
+                    _min.X = value.X;
+                }
```

Comment:
Suggest we use clamping instead of if/else for these.

Reply:
I don't exactly get what you mean by that. And clamping would have the same branch count...

Reply:
I tried making them work with clamping but i couldn't any suggestions on how to do this?